src/plugins/auto/auto_schedule.cpp

```diff
@@ -676,23 +678,23 @@ void AutoSchedule::WaitActualNetworkReady() const {
 bool AutoSchedule::ScheduleToWorkerInferRequest(IE::Task inferPipelineTask, DeviceName preferred_device) {
     std::vector<DeviceInformation> devices;
     // AUTO work mode
+    // Devices that fail infer will be removed from the priority list in the callback, need lock here
+    std::unique_lock<std::mutex> lock(_autoSContext->_fallbackMutex);
     if (!preferred_device.empty()) {
         if (_pCTPUTLoadContext) {
-            std::lock_guard<std::mutex> lock(_autoSContext->_fallbackMutex);
             devices = _autoSContext->_devicePriorities;
         } else {
             // if the device needed by customer is not ready, need to wait for it
             WaitActualNetworkReady();
             // the preferred_device should be the selected device in AUTO work mode
+            // or, if fallback enabled, the fallback device will not fit in the remote tensor as well
```

Comment:
How about `preferred_device` for remote tensor cannot work and fallback device is CPU?

Reply:
here is kind of confusing, when feeding a gpu remote tensor, and -d AUTO:vpux,cpu, currently we are throwing an exception directly, because I think with remote tensor, user are explicitly expecting a performance benefit. probably need to understand the real customer need, what's their expectation on this scenario.